agent/lib/kontena/fluentd/message_processor.rb

```diff
@@ -0,0 +1,85 @@
+require_relative '../helpers/rpc_helper'
+
+module Kontena
+  module Fluentd
+    class MessageProcessor
+      include Celluloid
+      include Celluloid::Notifications
+      include Kontena::Logging
+      include Kontena::Helpers::RpcHelper
+
+      finalizer :close
+
+      def initialize(queue)
+        @queue = queue
+        @containers = {}
+        @messages = 0
+        @running = false
+        subscribe('websocket:connected', :on_connect) # from master_info RPC
+        subscribe('websocket:disconnect', :on_disconnect)
+        async.report
+      end
+
+      def on_connect(_, _)
+        info "websocket connected"
+        async.start unless @running
```

Comment:
If this actor crashes and restarts, it will not `start` again until the websocket reconnects. TODO: websocket client observable :)